### DISCUSSION TITLE
This edge function has crashed error

### DISCUSSION THREAD

**Discussion by rosso007** [vercel/community#5192] [Help]
### Summary

![image](https://github.com/vercel/community/assets/4747855/132ba8d7-cfbe-4957-ac26-caf1f85cd8b6)

Im getting this error when building my app but im not getting any errors in the build logs.

I get the following errors in the logs post deploy:

[GET] [middleware: "middleware"] / reason=EDGE_FUNCTION_INVOCATION_FAILED, status=500, user_error=true

ReferenceError: __dirname is not defined
    at (node_modules/next/dist/compiled/ua-parser-js/ua-parser.js:1:17315)
    at (node_modules/next/dist/compiled/ua-parser-js/ua-parser.js:1:17444)
    at (node_modules/next/dist/server/web/spec-extension/user-agent.js:27:59)
    at (node_modules/next/server.js:8:0)
    at (middleware.js:38:29)
    at (middleware:middleware.js:1:17)
    
    

Im using Next.js 14 and supabase. i have a supabase middleware from here: https://supabase.com/docs/guides/auth/auth-helpers/nextjs?language=ts

deploy to vercel. Builds fine via npm run build in github codespaces and runs fine using npm run dev. No idea whats causing the issue? Anyone else that this problem? I can't seem to find the issue with the middleware.

Edge functions seem like a nightmare :(

Update 1:

I changed the runtimes to nodejs for all files which were building as edge, but still got the same error.

Update 2:

In this course from supabase, they deploy a twitter clone to vercel and its recorded over video working. Now we I deply the same code, I get the same edge function error. This has to be an error with vercel deployments? 

https://egghead.io/courses/build-a-twitter-clone-with-the-next-js-app-router-and-supabase-19bebadb

**Comment by naka0000** [comment]
I am also having same error using Next.js 14 and supabase.

**Comment by Im-HaGGling** [comment]
I also encountered this problem using the template - https://github.com/vercel/next.js/tree/canary/examples/app-dir-i18n-routing from documentation - https://nextjs.org/docs/app/building-your-application/routing/internationalization

```bash
ReferenceError: __dirname is not defined
    at (node_modules/next/dist/compiled/ua-parser-js/ua-parser.js:1:17315)
    at (node_modules/next/dist/compiled/ua-parser-js/ua-parser.js:1:17444)
    at (node_modules/next/dist/server/web/spec-extension/user-agent.js:27:59)
    at (node_modules/next/server.js:8:0)
    at (middleware.js:1:29)
```

middleware.ts:
```ts
import type {NextRequest} from "next/server"
import {NextResponse} from "next/server"
import Negotiator from "negotiator"
import {match as matchLocale} from "@formatjs/intl-localematcher"

import {i18n} from "./i18n-config"

function getLocale(request: NextRequest): string | undefined {
  // Negotiator expects plain object so we need to transform headers
  const negotiatorHeaders: Record<string, string> = {}
  request.headers.forEach((value, key) => (negotiatorHeaders[key] = value))

  // @ts-ignore locales are readonly
  const locales: string[] = i18n.locales

  // Use negotiator and intl-localematcher to get best locale
  const languages = new Negotiator({headers: negotiatorHeaders}).languages(
    locales,
  )

  const locale = matchLocale(languages, locales, i18n.defaultLocale)

  return locale
}

export function middleware(request: NextRequest) {
  const pathname = request.nextUrl.pathname

  // // `/_next/` and `/api/` are ignored by the watcher, but we need to ignore files in `public` manually.
  // // If you have one
  // if (
  //   [
  //     '/manifest.json',
  //     '/favicon.ico',
  //     // Your other files in `public`
  //   ].includes(pathname)
  // )
  //   return

  // Check if there is any supported locale in the pathname
  const pathnameIsMissingLocale = i18n.locales.every(
    (locale) =>
      !pathname.startsWith(`/${locale}/`) && pathname !== `/${locale}`,
  )

  // Redirect if there is no locale
  if (pathnameIsMissingLocale) {
    const locale = getLocale(request)

    // e.g. incoming request is /products
    // The new URL is now /en-US/products
    return NextResponse.redirect(
      new URL(
        `/${locale}${pathname.startsWith("/") ? "" : "/"}${pathname}`,
        request.url,
      ),
    )
  }
}

export const config = {
  // Matcher ignoring `/_next/` and `/api/`
  matcher: ["/((?!api|_next/static|_next/image|favicon.ico).*)"],
}

```

**Comment by Im-HaGGling** [comment]
I found the answer and solution here - https://nextjs-forum.com/post/1184005369006133298
@rosso007 @naka0000

**Comment by naka0000** [comment]
![deploy](https://github.com/vercel/community/assets/91176858/cb1764f2-9f79-489a-8411-4804f43084f4)

i found answer too.
you can deploy setting change.

Build & Development Settings -> Project Settings -> Output Directory
set `.next`
